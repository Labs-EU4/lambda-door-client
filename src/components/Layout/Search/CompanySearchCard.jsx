import React from 'react';
import styled from 'styled-components';
import { Card } from 'antd';

const CompanySearchCard = ({ history, name, location, website, type, id }) => {
  return (
    <StyledCard onClick={() => history.push(`/company-page/${id}`)}>
      <h2>{name}</h2>
      <p>{location}</p>
      <a href={website}>{website}</a>
      <p>
        Type: &nbsp;
        {type}
      </p>
    </StyledCard>
  );
};

export default CompanySearchCard;

const StyledCard = styled(Card)`
  height: 200px;
  width: 350px;
  margin-right: 1rem !important;
  margin-bottom: 1rem !important;
  &:hover {
    border: 2px solid #bb1333;
    cursor: pointer;
  }
`;
