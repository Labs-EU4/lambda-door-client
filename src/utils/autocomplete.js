/* eslint-disable react/jsx-no-undef */
import React, { useState } from 'react';
import { Form, Input, AutoComplete, Button } from 'antd';
import AddCompanyModal from '../components/AddReview/AddCompanyModal';

const { Option } = AutoComplete;

const renderOption = Item => {
  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <Option key={Item.name} text={Item.name}>
      <p>{Item.name}</p>
    </Option>
  );
};

const AutoCompleted = ({ label, dataSource, placeholder, onChange }) => {
  const [addingCompany, setAddingCompany] = useState(false);
  const [value, setValue] = useState('');

  const options = dataSource.map(renderOption).concat([
    <Option key="all" value="" className="show-all">
      <Button type="button" onClick={() => setAddingCompany(true)}>
        Add Company
      </Button>
    </Option>,
  ]);

  return (
    <>
      <AddCompanyModal
        visible={addingCompany}
        setAddingCompany={setAddingCompany}
        value={value}
      />
      <Form.Item label={label}>
        <AutoComplete
          className="certain-category-search"
          dropdownClassName="certain-category-search-dropdown"
          dropdownMatchSelectWidth={false}
          dropdownStyle={{ width: 300 }}
          size="default"
          style={{ width: '100%' }}
          dataSource={options}
          optionLabelProp="value"
          filterOption={(inputValue, option) => {
            if (
              option.key.toLowerCase().includes(inputValue.toLowerCase()) ||
              option.key.toLowerCase() === 'all'
            ) {
              return true;
            }
            return false;
          }}
          onChange={e => {
            if (e) {
              onChange(e);
              setValue(e);
            }
          }}
        >
          <Input size="default" placeholder={placeholder} />
        </AutoComplete>
      </Form.Item>
    </>
  );
};

export default AutoCompleted;
