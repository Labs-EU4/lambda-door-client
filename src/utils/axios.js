import axios from 'axios';

axios.defaults.baseURL = `${process.env.REACT_APP_BACKEND_URL}`

export default function withAuth() {
  const token = localStorage.getItem('token');
  const instance = axios.create({
    headers: {
      'Content-Type': 'application/json',
      Authorization: `${token}`,
    },
  });

  return instance;
}
