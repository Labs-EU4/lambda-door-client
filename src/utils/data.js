export const companies = [
  {
    id: 1,
    name: 'Accenture',
    website: 'www.accenture.com.',
    location: 'Atlanta, GA',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 2,
    name: 'Anthem, Inc.',
    website: 'https://www.antheminc.com',
    location: 'Atlanta, GA',
    type: 'Health',
    logo: '',
    description: '',
  },
  {
    id: 3,
    name: 'Ahalogy',
    website: 'https://www.ahalogy.com',
    location: 'Cincinnati, OH',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 4,
    name: 'Bad Rabbit',
    website: 'https://www.badrabbit.com',
    location: 'Portland, OR',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 5,
    name: 'DoNotPay Inc',
    website: 'https://donotpay.com',
    location: 'Technology',
    type: 'rowValue3',
    logo: '',
    description: '',
  },
  {
    id: 6,
    name: 'Newfront Insurance',
    website: 'https://www.newfrontinsurance.com',
    location: 'San Francisco, CA',
    type: 'insurance',
    logo: '',
    description: '',
  },
  {
    id: 7,
    name: 'Unspun',
    website: 'https://www.unspuntech.com/',
    location: 'San Francisco, CA',
    type: 'Commerce',
    logo: '',
    description: '',
  },
  {
    id: 8,
    name: 'rideOS',
    website: 'https://rideos.ai',
    location: 'San Francisco, CA',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 9,
    name: 'Mesa',
    website: '',
    location: 'Austin, TX',
    type: 'Software',
    logo: '',
    description: '',
  },
  {
    id: 10,
    name: 'Paystack',
    website: 'https://paystack.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 11,
    name: 'BuyCoins',
    website: 'https://buycoins.africa/',
    location: 'Lagos, NG',
    type: 'Crypto',
    logo: '',
    description: '',
  },
  {
    id: 12,
    name: 'Chipper Cash',
    website: 'https://chippercash.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 13,
    name: 'Cowrywise',
    website: 'https://www.cowrywise.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 14,
    name: 'Others',
    website: '',
    location: '',
    type: '',
    logo: '',
    description: '',
  },
];
export const salaryReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    description: 'Accenture Programmer',
    salary: 95000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 2,
    user_id: 2,
    company_id: 3,
    description: 'Software Engineer',
    salary: 98000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 3,
    user_id: 3,
    company_id: 10,
    description: 'Junior Developer',
    salary: 3000000,
    currency: 'NGN',
    is_accepting_questions: 1,
  },
  {
    id: 4,
    user_id: 4,
    company_id: 1,
    description: 'Accenture Technical Specialist',
    salary: 85000,
    currency: 'USD',
    is_accepting_questions: 1,
  },
  {
    id: 5,
    user_id: 5,
    company_id: 13,
    description: 'Software Engineer',
    salary: '4800000',
    currency: 'NGN',
    is_accepting_questions: 0,
  },
];
export const interviewReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    company_name: 'Microsoft',
    text:
      'Six rounds of phone/tech interviews over a long time period. It seemed a bit scattered and could have been way more efficient. I felt like some of the interviews got repetitive. The entire process took three months and I was able to interview at another company and find a job whilst waiting to hear back.',
  },
  {
    id: 2,
    user_id: 2,
    company_id: 5,
    company_name: 'Google',
    text:
      'Very pleasant recruiters. First a phone interview that took about 15 minutes. Followed by two Skype interviews; first with a recruiter after that with a manager.Basic trouble shooting questions. They want to really get to know you as a person. Just be yourself and do not be afraid to be geeky/nerdy! Make sure you have a desk and dedicated home phone line.',
  },
  {
    id: 3,
    user_id: 4,
    company_id: 11,
    company_name: 'Amazon',
    text:
      'Completed their developer challenge and received an email offering an interview. A few preliminary questions via email. Scheduled a phone interview. The recruiter did not call me, and did not respond to my email asking why they missed the scheduled interview. I am not sure what happened, but after scheduling the interview, there is no reason to not show up.',
  },
  {
    id: 4,
    user_id: 3,
    company_id: 6,
    company_name: 'Twitter',
    text:
      'There is one phone interview for an hour. If you clear this interview, you will be invited for an onsite interview. There are five interviews onsite, all in one day. Interviews are in a casual environment. After the first two interviews, you take a break and are escorted for lunch.',
  },
  {
    id: 5,
    user_id: 5,
    company_id: 12,
    company_name: 'Slack',
    text:
      "They initially called me in for a technical interview and I didn't hear anything back for three weeks. After this I was surprised to get another call saying that I had been invited for a behavioural interview. In total the process took over two months. The interviews were both with a panel and were very thorough but the atmosphere was relaxed and friendly.",
  },

  {
    company_id: 21,
    company_name: 'Novosco',
    text:
      'I was invited to a very informal chat in the first place and shown around the office. Two weeks later I got a technical interview and I met some more people from the office. The interview lasted for an hour and had both easy and difficult questions. Two days later I got a phone call saying that I had got the job.',
  },
];

export const allInterests = [
  'Software Engineer',
  'Front End',
  'Back End',
  'Full Stack',
  'Data Science',
  'Machine Learning',
  'User Experience',
  'Mobile Development',
  'Product Manager',
];
export const jobCategories = [
  {
    id: 1,
    name: 'Software Engineer',
  },
  {
    id: 2,
    name: 'Front End',
  },
  {
    id: 3,
    name: 'Back End',
  },
  {
    id: 4,
    name: 'Full Stack',
  },
  {
    id: 5,
    name: 'Data Science',
  },
  {
    id: 6,
    name: 'Machine Learning',
  },
  {
    id: 7,
    name: 'User Experience',
  },
];
export const currencies = [
  {
    id: 1,
    name: 'USD',
    symbol: '$',
    full_name: 'USD Dollars',
  },
  {
    id: 2,
    name: 'EUR',
    symbol: '',
    full_name: 'EUROS',
  },
  {
    id: 3,
    name: 'GBP',
    symbol: '',
    full_name: 'British Pounds',
  },
  {
    id: 4,
    name: 'NAIRA',
    symbol: '',
    full_name: 'Nigerian Naira',
  },
];
