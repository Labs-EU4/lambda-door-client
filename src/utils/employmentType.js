export const employmentType = [
  {
    id: 1,
    name: 'Full-time',
  },
  {
    id: 2,
    name: 'Contract',
  },
  {
    id: 3,
    name: 'Part-time',
  },
  {
    id: 4,
    name: 'Internship',
  },
];
