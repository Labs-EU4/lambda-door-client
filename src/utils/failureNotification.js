import { notification } from 'antd';

const failureNotification = message => {
  notification.open({
    message: 'Something went wrong',
    description: message,
  });
};

export default failureNotification;
