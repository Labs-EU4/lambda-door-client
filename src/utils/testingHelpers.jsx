import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import rootReducer from '../state/reducers/index';

export const TestingRouter = ({
  ComponentWithRedirection,
  RedirectUrl,
  MainRoute,
}) => (
  <>
    <Route
      path={MainRoute}
      exact={true}
      render={props => {
        return <ComponentWithRedirection {...props} />;
      }}
    />
    <Route path={RedirectUrl} render={() => <div>{RedirectUrl}</div>} />
  </>
);

export const renderWithRedux = (
  component,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      compose(applyMiddleware(thunk))
    ),
  } = {},
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>
    ),
    store,
    history,
  };
};
