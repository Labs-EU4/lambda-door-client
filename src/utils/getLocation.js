import axios from 'axios';
import store from '../state/store';
import { editProfile } from '../state/actions/user';

export async function showPosition(position, id) {
  const { longitude, latitude } = position.coords;

  const {
    data: { results },
  } = await axios.get(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&result_type=country&key=${process.env.REACT_APP_GOOGLE_API_KEY}`
  );
  await store.dispatch(
    editProfile(
      { longitude, latitude, location: results[0].formatted_address },
      id
    )
  );
}

function showError(error) {
  switch (error.code) {
    case error.PERMISSION_DENIED:
      console.log('User denied the request for Geolocation.');
      break;
    case error.POSITION_UNAVAILABLE:
      console.log('Location information is unavailable.');
      break;
    case error.TIMEOUT:
      console.log('The request to get user location timed out.');
      break;
    case error.UNKNOWN_ERROR:
      console.log('An unknown error occurred.');
      break;
    default:
      return null;
  }
  return null;
}
export function getLocation(id) {
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(
      position => showPosition(position, id),
      showError
    );
  }
}
