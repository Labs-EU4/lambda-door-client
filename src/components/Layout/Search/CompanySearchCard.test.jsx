import React from 'react';
import * as rtl from '@testing-library/react';
import CompanySearchCard from './CompanySearchCard';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from '../../../utils/testingHelpers';

beforeEach(rtl.cleanup);

describe('CompanySearchCard', () => {
  it('renders without crashing', () => {
    renderWithRedux(<CompanySearchCard />);
  });

  it('renders correctly', () => {
    expect(
      renderWithRedux(<CompanySearchCard />).baseElement
    ).toMatchSnapshot();
  });
});
