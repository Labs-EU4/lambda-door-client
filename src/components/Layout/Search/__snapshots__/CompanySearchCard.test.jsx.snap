// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySearchCard renders correctly 1`] = `
<body>
  <div>
    <div
      class="ant-card sc-bdVaJa lnrxHJ ant-card-bordered"
    >
      <div
        class="ant-card-body"
      >
        <h2 />
        <p />
        <a />
        <p>
          Type: Â 
        </p>
      </div>
    </div>
  </div>
</body>
`;
